src/main/java/net/mcreator/util/DesktopUtils.java

```diff
@@ -36,7 +37,8 @@ public static void browseSafe(String uri) {
 		try {
 			new Thread(() -> {
 				try {
-					browse(new URI(uri));
+					if (MCreatorApplication.isInternet)

```

Comment:
I think you should leave this out as this browse may be used for URLs outside MCreator domain. `MCreatorApplication.isInternet` is actually more like `MCreatorApplication.isMCreatorWebsiteUp`

Reply:
✔️ Fixed

Reply:
✔️ Fixed
Now it checks whether the _given_ website is up.

Reply:
@KlemenDEV False value of `WebIO.readURLToString(uri).equals("")` does not actually mean that MCreator website is inaccessible, this string is a replacement for `MCreatorApplication.isInternet` (as you asked).
Plus, I made it to not confuse users with an empty page opening in the browser after their actions inside the application.

Reply:
@KlemenDEV False value of `WebIO.readURLToString(uri).equals("")` does not actually mean that MCreator website is inaccessible, this string is a replacement for `MCreatorApplication.isInternet` (as you asked).
Plus, I made it to not confuse users without connection with an empty page opening in the browser after their actions inside the application.

Reply:
WebIO.readURLToString(uri).equals("") may not be able to access internet due to eg firewall but browser may still work, we can not assume no internet on PC from no internet for MCreator

Reply:
Oops, that's what you meant. Then it's okay.
Maybe sometime in the future checking for firewalls will be implemented too :)